:books: Refine error message to show how to install mmengine.

saratrajput opened the pull request:
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Refine error message to tell the user how to install mmengine.

## Modification

Modifies [the error message in mim](https://github.com/open-mmlab/mim/blob/ffaff8a4ece8a833b20dc7c6c1a7f8647d9eec15/mim/commands/download.py#L92) with:
```
Please install mmengine to use the download command: mim install mmengine.'
```

## BC-breaking (Optional)

Does the modification introduce changes that break the back-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

saratrajput commented:
Reference issue: https://github.com/open-mmlab/mmsegmentation/issues/2072

ice-tong commented on mim/commands/download.py:
```diff
@@ -89,7 +89,7 @@ def download(package: str,
         from mmengine import Config
     except ImportError:
-        msg = 'Please install mmengine to use the download command.'
+        msg = 'Please install mmengine to use the download command: mim install mmengine.'
```
```suggestion
        msg = 'Please install mmengine to use the download command: `mim install mmengine`.'
```

ice-tong replied:
This line is too long than 79 and suggests a new line.

ice-tong commented:
Hi @saratrajput , thanks for the contribution!

The gridsearch command have the same error message should be modified: https://github.com/open-mmlab/mim/blob/ffaff8a4ece8a833b20dc7c6c1a7f8647d9eec15/mim/commands/gridsearch.py#L275

And make sure you have pre-commit your code before committing. ^_^
```
pip install -U pre-commit
pre-commit install
git commit -m "xxx"  # this will trigger pre-commit
```

saratrajput commented:
@ice-tong Sorry for the late update. I've split the error message to avoid the linting errors, and updated the error message in `gridsearch.py` as well. 

The error message is displayed as shown in the below images.

![image](https://user-images.githubusercontent.com/29677848/194012414-a19ab2a8-39bb-42a0-8816-e98fcae1a77b.png)
![image](https://user-images.githubusercontent.com/29677848/194012850-72ff88ca-966d-4b77-a1aa-b808b5c54b95.png)

ice-tong commented:
Hi @saratrajput, thanks for the kind contribution!
Since the main branch has been changed, you should rebase the main branch and modify that again. ^_^

saratrajput commented:
@ice-tong Done.

ice-tong commented on mim/commands/download.py:
```diff
@@ -98,7 +98,8 @@ def download(package: str,
         except ImportError:
             raise ImportError(
-                'Please install mmengine to use the download command!')
+                '''Please install mmengine to use the download command:
+            `mim install mmengine`.''')
```
```suggestion
                'Please install mmengine to use the download command: '
                '`mim install mmengine`.')
```
I think we don't need to wrap a new line ~
![image](https://user-images.githubusercontent.com/32220263/194837509-336a4a7e-6693-4ffd-86af-b771d7cdc0fb.png)

ice-tong commented on mim/commands/gridsearch.py:
```diff
@@ -276,7 +276,8 @@ def gridsearch(
         except ImportError:
             raise ImportError(
-                'Please install mmengine to use the gridsearch command!')
+                '''Please install mmengine to use the gridsearch command:
+            `mim install mmengine`.''')
```
```suggestion
                'Please install mmengine to use the gridsearch command: '
                '`mim install mmengine`.')
```
same ~

ice-tong commented:
Hi @saratrajput , thanks for your contribution!

please @zhouzaida  have a look at this PR~
